Resources/config/services.xml

```diff
@@ -8,8 +8,31 @@
             <tag name="kernel.event_subscriber"/>
         </service>
 
-        <!-- enable autowiring compatibility in Symfony 3.3+ -->
-        <service id="Http\Client\HttpClient" alias="httplug.client" public="false" />
+        <!-- Discovery with autowiring support for Symfony 3.3+ -->
+        <service id="httplug.message_factory.default" class="Http\Message\MessageFactory">
+            <factory class="Http\Discovery\MessageFactoryDiscovery" method="find" />
+        </service>
+        <service id="Http\Message\MessageFactory" alias="httplug.message_factory.default" public="false" />
+
+        <service id="httplug.stream_factory.default" class="Http\Message\StreamFactory">
+            <factory class="Http\Discovery\StreamFactoryDiscovery" method="find" />
+        </service>
+        <service id="Http\Message\StreamFactory" alias="httplug.stream_factory.default" public="false" />
+
+        <service id="httplug.uri_factory.default" class="Http\Message\UriFactory">
+            <factory class="Http\Discovery\UriFactoryDiscovery" method="find" />
+        </service>
+        <service id="Http\Message\UriFactory" alias="httplug.uri_factory.default" public="false" />
```

Comment:
I'm 200% with this idea, but I think we have a lot of work around all those services. The main issue is probably the ability to override any service with the dependency injection container. I don't know if BC will be kept if we invert the real definition and the alias as a user may replace the old-style name by a definition (instead of an alias) and so break the equity between new-style = old-style. But can this be considered as a BC break?